reframe/core/containers.py

```diff
@@ -259,10 +260,19 @@ def launch_command(self, stagedir):
 
         run_opts += self.options
         if self.command:
-            return (f'singularity exec {" ".join(run_opts)} '
+            return (f'{self._launch_command} exec {" ".join(run_opts)} '
                     f'{self.image} {self.command}')
 
-        return f'singularity run {" ".join(run_opts)} {self.image}'
+        return f'{self._launch_command} run {" ".join(run_opts)} {self.image}'
+
+
+class Apptainer(Singularity):
+    '''Container platform backend for running containers with `Apptainer
+    <https://apptainer.org/>`__.'''
```

Comment:
Add an annotation here that this was added in version 4.0.0.

Reply:
You need to list it also here: https://reframe-hpc.readthedocs.io/en/stable/config_reference.html#systems-.partitions-.container_platforms-.type